src/ast/LlvmVisitor.java

```diff
@@ -109,67 +163,232 @@ public void visit(MainClass mainClass) {
-        LlvmProgram = LlvmProgram.concat("\t%" + varDecl.name() + " = alloca " + getLlvmType(varDecl) + "\n");
+        // note that this method would never be called on a class's field, only local var
+        // note 2: in minijava there's no 'int x = 10;' just the declaration 'int x;' so no need to load & store right now
+        LLVMProgram = LLVMProgram.concat("\t%" + varDecl.name() + " = alloca " + getLLVMType(varDecl) + "\n");
     }
 
     @Override
     public void visit(BlockStatement blockStatement) {
-        for(int i = 0; i < blockStatement.statements().size(); i++) blockStatement.statements().get(i).accept(this);
+        for (int i = 0; i < blockStatement.statements().size(); i++) {
+            blockStatement.statements().get(i).accept(this);
+        };
     }
 
     @Override
     public void visit(IfStatement ifStatement) {
-
+        ifStatement.cond().accept(this);
+        LLVMProgram = LLVMProgram.concat(String.format(
+                "\tbr i1 %%_%d, label %%if%d, label %%if%d\n", methodCurrRegIndex -1, methodCurrLabelIndex, methodCurrLabelIndex +1));
+        LLVMProgram = LLVMProgram.concat(String.format(
+                "if%d:\n", methodCurrLabelIndex++
+        ));
+        ifStatement.thencase().accept(this);
+        LLVMProgram = LLVMProgram.concat(String.format(
+                "if%d:\n", methodCurrLabelIndex++
+        ));
+        ifStatement.elsecase().accept(this);
     }
 
     @Override
     public void visit(WhileStatement whileStatement) {
-
+        LLVMProgram = LLVMProgram.concat(String.format(
+                "\tbr label %%while_cond%d\n", methodCurrLabelIndex
+        ));
+        LLVMProgram = LLVMProgram.concat(String.format(
+                "while_cond%d:\n", methodCurrLabelIndex++
+        ));
+        whileStatement.cond().accept(this);
+        LLVMProgram = LLVMProgram.concat(String.format(
+                "\tbr i1 %%_%d, label %%while_loop%d, label %%while_end%d\n", methodCurrRegIndex -1, methodCurrLabelIndex, methodCurrLabelIndex +1
+        ));
+        LLVMProgram = LLVMProgram.concat(String.format(
+                "while_loop%d:\n", methodCurrLabelIndex++
+        ));
+        whileStatement.body().accept(this);
+        LLVMProgram = LLVMProgram.concat(String.format(
+                "while_end%d:\n", methodCurrLabelIndex++
+        ));
     }
 
     @Override
     public void visit(SysoutStatement sysoutStatement) {
-
+        sysoutStatement.arg().accept(this);
+        LLVMProgram = LLVMProgram.concat(String.format(
+                "\tcall void (i32) @print_int(i32 %%_%d)\n", methodCurrRegIndex-1
+        ));
     }
 
     @Override
     public void visit(AssignStatement assignStatement) {
-
+        String assigneeName = assignStatement.lv();
+        SymbolTable enclosingMethodST = assignStatement.enclosingScope();
+        VariableIntroduction assigneeVarIntro = (VariableIntroduction) enclosingMethodST.getSymbol(assigneeName, false).declaration();
```

Comment:
use STLookup.findDeclTable() instead of getSymbol to avoid null pointer exception